compiler+runtime/src/cpp/jank/runtime/obj/integer_range.cpp

```diff
@@ -0,0 +1,193 @@
+#include <jank/runtime/obj/integer_range.hpp>
+#include <jank/runtime/obj/number.hpp>
+#include <jank/runtime/core/math.hpp>
+#include <jank/runtime/core/make_box.hpp>
+#include <jank/runtime/visit.hpp>
+#include <jank/runtime/behavior/metadatable.hpp>
+
+namespace jank::runtime
+{
+  static native_bool positive_step_bounds_check(native_integer const val, native_integer const end)
+  {
+    return end <= val;
+  }
+
+  static native_bool negative_step_bounds_check(native_integer const val, native_integer const end)
+  {
+    return val <= end;
+  }
+
+  obj::integer_range::static_object(native_integer const end)
+    : end{ end }
+    , step{ 1 }
+    , bounds_check{ positive_step_bounds_check }
+  {
+  }
+
+  obj::integer_range::static_object(native_integer const start, native_integer const end)
+    : start{ start }
+    , end{ end }
+    , step{ 1 }
+    , bounds_check{ positive_step_bounds_check }
+  {
+  }
+
+  obj::integer_range::static_object(native_integer const start,
+                                    native_integer const end,
+                                    native_integer const step)
+    : start{ start }
+    , end{ end }
+    , step{ step }
+    , bounds_check{ step > 0 ? positive_step_bounds_check : negative_step_bounds_check }
+  {
+  }
+
+  obj::integer_range::static_object(native_integer const start,
+                                    native_integer const end,
+                                    native_integer const step,
+                                    obj::integer_range::bounds_check_t const bounds_check)
+    : start{ start }
+    , end{ end }
+    , step{ step }
+    , bounds_check{ bounds_check }
+  {
+  }
+
+  object_ptr obj::integer_range::create(native_integer const end)
+  {
+    if(end)
```

Comment:
There's a bug here, with `if(end)`. Can you reason through it?

Reply:
I think 'end' should be positive only, so that would be an issue here because in Clojure JVM -> (range -10) = (). So fixing this to be "if (end >0)"